main.py

```diff
@@ -1,44 +1,60 @@
-from re import search
+from flask import Flask, render_template
 from dotenv import dotenv_values
 from random import choice
 import urllib.parse
-import webbrowser
 import requests
 import json
+import os
+
+app = Flask(__name__)
 
 config = dotenv_values('.env')
 OPENCAGEDATA_API_KEY = config.get('OPENCAGEDATA_API_KEY')
 GEOAPIFY_API_KEY = config.get('GEOAPIFY_API_KEY')
 
-def randomCity():
+POPULATION_THRESHHOLD = 400000
+
+class City:
+    def __init__(self, name, coords):
+        self.name = name 
+        self.coords = coords 
+
+def getRandomCityName(populationThreshold):
     with open('city_data/worldCities.json', 'r', encoding='UTF-8') as r:
-        data = json.loads(r.read())
-        rand_city = choice(data)
-        
-        city = str(rand_city['city']).replace('"', '')
-        print(f"\n{city}'s population is {rand_city['population']}\n")
-        print(f"Country: {rand_city['country']}\n\nState/Region/Province: {rand_city['admin_name']}")
+        cities = json.loads(r.read())
+        nonEmptyPopulation = list(filter(lambda city: (city['population'] != ""), cities))
+        filteredCities = list(filter(lambda city: (city['population'] > populationThreshold), nonEmptyPopulation))
+        randCity = choice(filteredCities)
         
+        city = str(randCity['city']).replace('"', '')
         return city 
 
-def searchCity(city):
-    geo_url = 'https://api.opencagedata.com/geocode/v1/json'
+def getCityCoordinatesByName(cityName):
+    geoUrl = 'https://api.opencagedata.com/geocode/v1/json'
     params = {
-        'q': city,
+        'q': cityName,
         'key': OPENCAGEDATA_API_KEY
     }
-    r = requests.get(geo_url, params=params)
-    res = json.loads(r.text)
+    r = requests.get(geoUrl, params=params)
+    res = json.loads(r.text)['results'][0]['geometry']
     
-    lat = res['results'][0]['geometry']['lat']
-    lon = res['results'][0]['geometry']['lng']
+    lat = res['lat']
+    lon = res['lng']
     
-    with open('city_data/testData.json', 'w', encoding='UTF-8') as f:
-        json.dump(res, f, ensure_ascii=False, indent=4)
-        
-    return lat, lon
+    # with open('city_data/testData.json', 'w', encoding='UTF-8') as f:
+    #     json.dump(res, f, ensure_ascii=False, indent=4)
+    return (lon, lat)
+
+def getRandomCity():
+    cityName = getRandomCityName(POPULATION_THRESHHOLD)
+    cityCoords = getCityCoordinatesByName(cityName)
+    city = City(cityName, cityCoords)
+    print(f'{city.name} : {city.coords}')
+    
+    return city 
 
-def showMap(lat, lon):
+def showMap(city):
+    lon, lat = city.coords
```

Comment:
hnnng now this is podracing!

Reply:
All that work and talks to make sure destructuring a gaht dang tuple looked good. And it was worth it